### PR TITLE
Add migration dir to `PartialCardanoConfiguration`

### PR DESCRIPTION
**Pull request by Jimbo4350**
In order for the `cardano-explorer` to use the POM design, an additional field is needed in `PartialCardanoConfiguration` & `CardanoConfiguration`.
Related PR: https://github.com/input-output-hk/cardano-explorer/pull/121

### CODE REVIEW

File: cardano-config/src/Cardano/Config/Presets.hs [hunk 1]
```diff
@@ -39,6 +39,7 @@ mainnetConfiguration =
     , pccLogConfig = pure "./configuration/log-configuration.yaml"
     , pccDBPath = pure "./db/"
     , pccApplicationLockFile = pure ""
+    , pccMigrationDir = pure ""
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-config/src/Cardano/Config/Presets.hs]: @erikd Suggestions for a sensible default?

[the remark this answers is not in the document]
Comment (erikd) [reply cardano-config/src/Cardano/Config/Presets.hs]: I thing they should be `Maybe FilePath` with a default of `Nothing`

[the remark this answers is not in the document]
Comment (erikd) [reply cardano-config/src/Cardano/Config/Presets.hs]: To have the node include a migration directory when the node has no use for it is not right.

I actually think we have this around the wrong way. How much of the node configuration does the explorer need to provide to the node? Maybe it makes more sense to actually have the explorer provide a full node configuration.

**Comment by Jimbo4350** [comment]
This is no longer necessary because we are no longer using the POM for launching the `node`. See https://github.com/input-output-hk/cardano-node/pull/289

**Pull request closed without merging**